src/installer/sources.py

```diff
@@ -138,6 +157,44 @@ def read_dist_info(self, filename: str) -> str:
         path = posixpath.join(self.dist_info_dir, filename)
         return self._zipfile.read(path).decode("utf-8")
 
+    def validate_record(self) -> None:
```

Comment:
`RecordEntry(...).validate(...)` seems a little bit expensive (as it basically requires extracting the whole wheel file instead of just accessing info and `RECORD`) Maybe allowing specifying whether to validate files within `RECORD` is okay?